podcastfy/client.py

```diff
@@ -5,121 +5,173 @@
 from URLs or existing transcript files. It orchestrates the content extraction,
 generation, and text-to-speech conversion processes.
 """
+import copy
 
 import os
 import uuid
 import typer
 import yaml
+
+from podcastfy.aiengines.llm.gemini_langchain import DefaultPodcastifyTranscriptEngine
+from podcastfy.aiengines.tts.base import TTSBackend
+from podcastfy.aiengines.tts.tts_backends import OpenAITTS, ElevenLabsTTS, EdgeTTS
+from podcastfy.core.audio import AudioManager
+from podcastfy.core.character import Character
+from podcastfy.core.content import Content
+from podcastfy.core.podcast import Podcast
+from podcastfy.core.transcript import Transcript
 from podcastfy.content_parser.content_extractor import ContentExtractor
-from podcastfy.content_generator import ContentGenerator
-from podcastfy.text_to_speech import TextToSpeech
+from podcastfy.core.tts_configs import TTSConfig
 from podcastfy.utils.config import Config, load_config
 from podcastfy.utils.config_conversation import (
-    ConversationConfig,
     load_conversation_config,
 )
 from podcastfy.utils.logger import setup_logger
 from typing import List, Optional, Dict, Any
-import copy
-
 
 logger = setup_logger(__name__)
 
 app = typer.Typer()
 
+def create_characters(config: Dict[str, Any]) -> List[Character]:
+    # in the future, we should load this from the config file
+    host = Character(
+        name="Person1",
+        role="Podcast host",
+        tts_configs={
+            "openai": TTSConfig(
+                voice=config["text_to_speech"]["openai"]["default_voices"]["question"],
+                backend="openai",
+            ),
+            "elevenlabs": TTSConfig(
+                voice=config["text_to_speech"]["elevenlabs"]["default_voices"][
+                    "question"
+                ],
+                backend="elevenlabs",
+            ),
+        },
+        default_description_for_llm="{name} is an enthusiastic podcast host. Speaks clearly and engagingly.",
+    )
+
+    guest = Character(
+        name="Person2",
+        role="Expert guest",
+        tts_configs={
+            "openai": TTSConfig(
+                voice=config["text_to_speech"]["openai"]["default_voices"]["answer"],
+                backend="openai",
+            ),
+            "elevenlabs": TTSConfig(
+                voice=config["text_to_speech"]["elevenlabs"]["default_voices"][
+                    "answer"
+                ],
+                backend="elevenlabs",
+            ),
+        },
+        default_description_for_llm="{name} is an expert guest. Shares knowledge in a friendly manner.",
+    )
+
+    return [host, guest]
+
+
+def create_tts_backends(config: Config) -> List[TTSBackend]:
```

Comment:
Why should we instantiate all available backends and then later filter out all but the one the user wants? Shouldn't we instead only instantiate what the user needs? I recommend a TTS Factory design pattern.